node/hyperbahn-client.js

```diff
@@ -380,10 +380,8 @@ function unadvertise(opts) {
     timers.clearTimeout(self._advertisementTimer);
     timers.clearTimeout(self.advertisementTimeoutTimer);
     self.latestAdvertisementResult = null;
-    self.state = States.UNADVERTISED;
-    self.emit('unadvertised');
     function unadvertiseInternalCb(error, result) {
-        if (error) {
+        if (error && error.type !== 'tchannel.connection.reset') {
```

Comment:
:-1: wrong fix.

Please fix Hyperbahn backend to drain connection rather then close.